util/content.go

```diff
@@ -73,3 +78,44 @@ func FindCIDType(ctx context.Context, root cid.Cid, dserv ipld.NodeGetter) (cont
 	}
 	return
 }
+
+func removeEmptyStrings(strList []string) []string {
+	var strListNoEmpty []string
+	for _, str := range strList {
+		if str != "" {
+			strListNoEmpty = append(strListNoEmpty, str)
+		}
+	}
+	return strListNoEmpty
+}
+
+// DirsFromPath splits a path into a list of directories
+func DirsFromPath(collectionPath string, filename string) ([]string, error) {
+	collectionPath = filepath.Clean(collectionPath)
+	if dir, file := filepath.Split(collectionPath); file == filename { // path ends in the filename
+		collectionPath = dir // only keep the part with dirs
+	}
+	dirs := strings.Split(collectionPath, "/")
+	dirs = removeEmptyStrings(dirs)
+	return dirs, nil
+}
+
+func EnsurePathIsLinked(dirs []string, rootNode *merkledag.ProtoNode, ds format.DAGService) (*merkledag.ProtoNode, error) {
+	lookupNode := rootNode
```

Comment:
why do this?

Reply:
Changing variable names so it's obvious that we start with the root node and lookup downwards